user_auth/views.py

```diff
@@ -0,0 +1,154 @@
+from django.contrib import messages
+from django.contrib.auth import authenticate
+from django.contrib.auth import login as log_in
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth.tokens import default_token_generator
+from django.core.mail import BadHeaderError, send_mail
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import redirect, render
+from django.template.loader import render_to_string
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.utils.safestring import mark_safe
+
+from .forms import (
+    NewPasswordForm,
+    ResetPasswordForm,
+    UserLoginForm,
+    UserSignUpForm,
+)
+from .models import User
+
+
+def signup(request: HttpRequest):
+    if request.user.is_authenticated:
+        return redirect("/main/")
+    if request.method == "POST":
+        form = UserSignUpForm(request.POST)
+        if form.is_valid():
+            user = form.save(commit=False)
+            user.password = make_password(form.cleaned_data["password"])
+            user.save()
+            messages.success(
+                request,
+                mark_safe("Account Created: <a href='/login'>Login?</a>"),
+            )
+            return redirect("/")
+        else:
```

Comment:
no need for the else, since you are doing return above it

Reply:
this else statement is for adding error msgs if the form is not valid.